docs/taskruns.md

```diff
@@ -698,11 +699,25 @@ object(s), if present. Any `Request` or `Limit` specified by the user (on `Task`
 
 For more information, see the [`LimitRange` support in Pipeline](./compute-resources.md#limitrange-support).
 
+### Specifying `Retries`
+You can use the `retries` field to set how many times you want to retry on a failed TaskRun.
+All TaskRun failures are retriable except for `Cancellation`.
+
+For a retriable `TaskRun`, when an error occurs:
+- The error status is archived in `.status.RetriesStatus`
+- The `Succeed` condition in `.status` is updated: 
+```
+Type: Succeed
+Status: Unknown
+Reason: CompleteWithRetries
```

Comment:
I'm also confused about this, but I think you mean that the TaskRun is still running but a retry was completed, is that right? If so, maybe reason `RetryCompleted` is clearer?

```yaml
Type: Succeed
Status: Unknown
Reason: RetryCompleted
```

We can even make it explicit by using reason `RetryFailed`

```yaml
Type: Succeed
Status: Unknown
Reason: RetryFailed
```

Reply:
And because we are using it as an intermediate status to signal that this TaskRun is going to be retried. I'm leaning towards `ToBeRetried`.

cc @pritidesai

Reply:
Either `Retrying` or `ToBeRetried` sounds good to me, thanks for the clarification!

Reply:
+1 for `Retrying`!

Reply:
Ah thank you @XinruZhang @jerop, it is clearer for me now. I do agree with either `Retrying` or `ToBeRetried` 🙃.

Reply:
I have no preferences on the name and not good at it anyways 🙃 

This `status` describes the `taskRun` either exited with non-zero or timed out with `N` numbers of retries left.

Reply:
@XinruZhang I see that you've updated the code to use reason `Retrying`, please update the docs here as well

Reply:
Update: I'm changing the `Reason` from `Retrying` to `ToBeRetried`. See https://github.com/tektoncd/community/pull/882#discussion_r1037209150 for detailed reason.